### PR TITLE
VPC ID Filter is added when Subnet Name is specified in Datasource

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM1**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #4872 

Output from acceptance testing:

```
resource "ibm_is_vpc" "testacc_vpc" {
	name = "sunitha-vpc"
}

resource "ibm_is_subnet" "testacc_subnet" {
	name = "sunitha-subnet"
	vpc = "${ibm_is_vpc.testacc_vpc.id}"
	zone = "us-south-1"
	total_ipv4_address_count = 16
}
data "ibm_is_subnet" "ds_subnet" {
	identifier = "${ibm_is_subnet.testacc_subnet.id}"
}

data "ibm_is_subnet" "ds_subnet_name" {
  depends_on = [ ibm_is_subnet.testacc_subnet ]
	name =   "sunitha-subnet"
}

data "ibm_is_subnet" "ds_subnet_vpcid_subentname" {
  depends_on = [ ibm_is_subnet.testacc_subnet ]
  name =   "sunitha-subnet"
  vpc =  "${ibm_is_vpc.testacc_vpc.id}"
}
```
Screenshot 2023-10-24 at 9 14 37 PM

Screenshot 2023-10-24 at 9 15 02 PM

### CODE REVIEW

File: ibm/service/vpc/data_source_ibm_is_subnet.go [hunk 1]
```diff
@@ -26,6 +26,13 @@ func DataSourceIBMISSubnet() *schema.Resource {
 				ValidateFunc: validate.InvokeDataSourceValidator("ibm_is_subnet", "identifier"),
 			},
 
+			// "vpc_id": {
```
Comment (sreekarbvibm) [inline ibm/service/vpc/data_source_ibm_is_subnet.go:29]: Please remove commented code

**Review by sreekarbvibm — APPROVED** [review]
Please remove commented code

Verified in us-south region

**Comment by sumitkumartiwari** [comment]
Please resolve the comments and create an upstream PR and add it here..

**Comment by SunithaGudisagarIBM1** [comment]
```
	data "ibm_is_subnets" "ds_subnets_resource_group" {
		/* depends_on = [ibm_is_subnet.testacc_subnet] */
		/* vpc = ibm_is_vpc.testacc_vpc.id */
    vpc_name = "tfp-vpc"
    		/* resource_group = "11caaa983d9c4beb82690daab08717e9" */

	}
```
Screenshot 2023-10-30 at 12 01 31 PM

**Comment by SunithaGudisagarIBM1** [comment]
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4892

**Pull request closed without merging**